cpp/include/raft/neighbors/brute_force.cuh

```diff
@@ -153,7 +155,8 @@ void knn(raft::device_resources const& handle,
          value_int k,
```

Comment:
Can we remove k from here? We have been removing it from the Ann APIs too. It doesn't make any sense to have it with mdspan because we can infer it based on the output arrays.

Reply:
done w/ https://github.com/rapidsai/raft/pull/1371/commits/78f7b1f6b0ba133b0d50bc0340ecc5759d0ebcf8